includes/payments/class-edd-payment.php

```diff
@@ -1147,7 +1147,16 @@ public function add_download( $download_id = 0, $args = array(), $options = arra
 			$subtotal -= round( $tax, edd_currency_decimal_filter() );
 		}
 
-		$total      = $subtotal - $discount + $tax;
+		$fees = 0;
+		if ( isset( $args['fees'] ) && ! empty( $args['fees'] ) ) {
```

Comment:
Ideally this should be
`if ( ! empty( $args['fees'] ) && is_array( $args['fees'] ) ) {`

since empty = isset() && !== false, therefore a separate isset isn't needed, though we do need it to be an array to use foreach

Reply:
agreed